src/Cli/dotnet/Program.cs

```diff
@@ -204,34 +225,52 @@ internal static int ProcessArgs(string[] args, ITelemetry telemetryClient = null
                 Console.WriteLine($"Telemetry is: {(telemetryClient.Enabled ? "Enabled" : "Disabled")}");
             }
 
+            PerformanceLogEventSource.Log.TelemetrySendIfEnabledStart();
             TelemetryEventEntry.SendFiltered(topLevelCommandParserResult);
+            PerformanceLogEventSource.Log.TelemetrySendIfEnabledStop();
 
             int exitCode;
             if (BuiltInCommandsCatalog.Commands.TryGetValue(topLevelCommandParserResult.Command, out var builtIn))
             {
+                PerformanceLogEventSource.Log.BuiltInCommandParserStart();
                 var parseResult = Parser.Instance.ParseFrom($"dotnet {topLevelCommandParserResult.Command}", appArgs.ToArray());
+                PerformanceLogEventSource.Log.BuiltInCommandParserStop();
+
                 if (!parseResult.Errors.Any())
                 {
+                    PerformanceLogEventSource.Log.TelemetrySendIfEnabledStart();
```

Comment:
this is a logging of a logging. TelemetryEventEntry.SendFiltered(parseResult) will raise an event then exit

Reply:
My goal here is to make sure that sending the Telemetry event doesn't take any appreciable period of time.  Are you saying that this method just saves the event in-process and then returns, and that the background thread will pick it up and actually send it?

---
In reply to: [505726958](https://github.com/wli3/sdk/pull/1#discussion_r505726958) [](ancestors = 505726958)

Reply:
[pretty much](https://github.com/dotnet/sdk/blob/9efb9e1b345f3bac6c9e42ecdc8d3ee7f22bba7f/src/Cli/dotnet/Telemetry/Telemetry.cs#L88)

Reply:
Gotcha.  So ideally, this really turns into something that is never large.  The cost of emitting these events is low, and so I'm inclined to include them for completeness - I'd like to be able to say that each action that happens on the main thread is covered by a set of events.  Any concerns with this?

Reply:
It would be a problem on the consumer side (who run the query and do the analysis). I don't know a good way to document what the event is actually measuring especially when someone has no context of the implementation detail of CLI. All they see is the event name. It happened few times from the existing log, and people make wrong business conclusion from it.

A workaround would be make the the event name very long so that it is not easily assumed. SendToSeparateWorkerThreadEventIfEnabledStart. It is ridiculous to look at, but it will also make people pause and think.

Reply:
Is your thought that people will look at a log and not know what some of the things mean, or turn these on when capturing a trace?  If so, I try not to worry about about the naming too much.  Generally, you have to have code context as well to understand how to interpret an event.  That said, totally on-board to consider tweaking names to be clearer.

Reply:
We also made it easy for the customer to read these logs. People could easily get "sending logs are super fast" conclusion by looking at the result, in fact it is at least 100ms due to web API call. But if it is opt-in, and we don't advertise people to read it, that should be fine